build/yamls/elastiflow/logstash/logstash.conf

```diff
@@ -0,0 +1,1595 @@
+input {
+  # IPFIX
+  udp {
+    id => "input_udp_ipfix_ipv4"
+    host => "${ELASTIFLOW_IPFIX_UDP_IPV4_HOST:0.0.0.0}"
+    port => "${ELASTIFLOW_IPFIX_UDP_IPV4_PORT:4739}"
+    workers => "${ELASTIFLOW_IPFIX_UDP_WORKERS:4}"
+    queue_size => "${ELASTIFLOW_IPFIX_UDP_QUEUE_SIZE:2048}"
+    receive_buffer_bytes => "${ELASTIFLOW_IPFIX_UDP_RCV_BUFF:33554432}"
+    codec => netflow {
+      versions => [10]
+      target => "ipfix"
+      include_flowset_id => "true"
+      cache_ttl => 86400
+      ipfix_definitions => "${ELASTIFLOW_DEFINITION_PATH:/etc/logstash/elastiflow/definitions}/ipfix.yml"
+    }
+    type => "ipfix"
+  }
+}
+filter {
+  # Initialize @metadata values used to control processing flow.
+  mutate {
+    id => "init_control"
+    add_field => {
+      "[@metadata][resolve_ip2host]" => "${ELASTIFLOW_RESOLVE_IP2HOST:false}"
+      "[@metadata][geoip_lookup]" => "${ELASTIFLOW_GEOIP_LOOKUP:true}"
+      "[@metadata][asn_lookup]" => "${ELASTIFLOW_ASN_LOOKUP:true}"
+      "[@metadata][keep_orig_data]" => "${ELASTIFLOW_KEEP_ORIG_DATA:true}"
+      "[node][ipaddr]" => "%{host}"
+      "[node][hostname]" => "%{host}"
+    }
+    replace => { "@version" => "3.5.3" }
+  }
+
+  # Add Logstash hostname.
+  ruby {
+    init => "
+      require 'socket'
+      @@hostname = Socket.gethostname
+    "
+    code => "
+      event.set('logstash_host', @@hostname)
+    "
+  }
+
+  # Adjust data for Netflow and IPFIX being received on the same port.
+  if [netflow][version] == 10 {
+    mutate {
+      rename => { "netflow" => "ipfix" }
+      replace => { "type" => "ipfix" }
+    }
+  } else if [ipfix][version] == 9 {
+    mutate {
+      rename => { "ipfix" => "netflow" }
+      replace => { "type" => "netflow" }
+    }
+  }
+
+  mutate {
+    id => "rename_host_type"
+    rename => {
+      "host" => "[event][host]"
+      "type" => "[event][type]"
+    }
+  }
+
+  # Resolve node IP address to hostname.
+  if [@metadata][resolve_ip2host] == "true" or [@metadata][resolve_ip2host] == "exporters" {
+    dns {
+      id => "dns_node_name"
+      reverse => [ "[node][hostname]" ]
+      action => "replace"
```

Comment:
Is this code added by us? I am asking because this code is before the license header of elastiflow repo. Should we treat the placement like that?

Reply:
Most of the code is from Elastiflow [config](https://github.com/robcowart/elastiflow/tree/master/logstash/elastiflow/conf.d) except input part (line 1-19). The filter and output part work for processing data to integrate with Kibana dashboard. I will suggest to keep as it is if it does not affect our functionality.

Reply:
Also thanks for mentioning the license header,  I will need to figure out whether we need to keep it or change.

Reply:
It was a question from me whether we differentiate the code added by us and code in Elastiflow repo by the license header.